Rakefile

```diff
@@ -17,6 +17,8 @@ Rake::TestTask.new(:test) do |t|
 end
 
 Rake::TestTask.new(:bench) do |t|
+  # Disable parallel tests since it can interfere with benchmark results and allocation counts
+  ENV["MT_CPU"] = "1"
```

Comment:
Minitest was parallelizing our benchmarks by default. This was the source of the extra random allocations.

https://github.com/seattlerb/minitest/blob/master/lib/minitest.rb#L27
https://github.com/seattlerb/minitest/blob/master/lib/minitest.rb#L138-L145